mlrun/api/utils/clients/log_collector.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import mlrun.api.utils.clients.protocols.grpc
+import mlrun.errors
+import mlrun.utils.singleton
+from mlrun.utils import logger
+
+
+class LogCollectorClient(
+    mlrun.api.utils.clients.protocols.grpc.BaseGRPCClient,
+    metaclass=mlrun.utils.singleton.Singleton,
+):
+    name = "log_collector"
+
+    def __init__(self, address: str = None):
+        self._initialize_proto_client_imports()
+        self.stub_class = self._log_collector_pb2_grpc.LogCollectorStub
+        super().__init__(address=address or mlrun.mlconf.log_collector.address)
+
+    def _initialize_proto_client_imports(self):
+        # Importing the proto client classes here and not at the top of the file to avoid raising an import error
+        # when the log_collector service is not enabled / the proto client wasn't compiled
+        import mlrun.api.proto.log_collector_pb2
+        import mlrun.api.proto.log_collector_pb2_grpc
+
+        self._log_collector_pb2 = mlrun.api.proto.log_collector_pb2
```

Comment:
Not a problem, but what does `pb2` mean here?

Reply:
thats how the protobuf compiler generates the files, I guess it is the protobuf protocol ? 🤔

Reply:
Yup